components/Tabs/tests/unit/Theming/TabsThemingTests.m

```diff
@@ -0,0 +1,197 @@
+  // When
+  [self.tabBar applyPrimaryThemeWithScheme:self.containerScheme];
+
+  // Then
+  [self verifyTabBarPrimaryTheming];
+}
+
+- (void)testTabBarPrimaryThemingCustom {
+  // When
+  self.colorScheme.primaryColor = [UIColor colorWithWhite:(CGFloat)0.9 alpha:0];
+  self.colorScheme.primaryColorVariant = [UIColor colorWithWhite:(CGFloat)0.8 alpha:(CGFloat)0.1];
+  self.colorScheme.secondaryColor = [UIColor colorWithWhite:(CGFloat)0.7 alpha:(CGFloat)0.2];
+  self.colorScheme.errorColor = [UIColor colorWithWhite:(CGFloat)0.6 alpha:(CGFloat)0.3];
+  self.colorScheme.surfaceColor = [UIColor colorWithWhite:(CGFloat)0.5 alpha:(CGFloat)0.4];
+  self.colorScheme.backgroundColor = [UIColor colorWithWhite:(CGFloat)0.4 alpha:(CGFloat)0.5];
+  self.colorScheme.onPrimaryColor = [UIColor colorWithWhite:(CGFloat)0.3 alpha:(CGFloat)0.6];
+  self.colorScheme.onSecondaryColor = [UIColor colorWithWhite:(CGFloat)0.2 alpha:(CGFloat)0.7];
+  self.colorScheme.onSurfaceColor = [UIColor colorWithWhite:(CGFloat)0.1 alpha:(CGFloat)0.8];
+  self.colorScheme.onBackgroundColor = [UIColor colorWithWhite:0 alpha:(CGFloat)0.9];
+
+  self.typographyScheme.headline1 = [UIFont systemFontOfSize:1];
+  self.typographyScheme.headline2 = [UIFont systemFontOfSize:2];
+  self.typographyScheme.headline3 = [UIFont systemFontOfSize:3];
+  self.typographyScheme.headline4 = [UIFont systemFontOfSize:4];
+  self.typographyScheme.headline5 = [UIFont systemFontOfSize:5];
+  self.typographyScheme.headline6 = [UIFont systemFontOfSize:6];
+  self.typographyScheme.subtitle1 = [UIFont systemFontOfSize:7];
+  self.typographyScheme.subtitle2 = [UIFont systemFontOfSize:8];
+  self.typographyScheme.body1 = [UIFont systemFontOfSize:9];
+  self.typographyScheme.body2 = [UIFont systemFontOfSize:10];
+  self.typographyScheme.caption = [UIFont systemFontOfSize:11];
+  self.typographyScheme.button = [UIFont systemFontOfSize:12];
+  self.typographyScheme.overline = [UIFont systemFontOfSize:13];
+
+  self.containerScheme.colorScheme = self.colorScheme;
+  self.containerScheme.typographyScheme = self.typographyScheme;
+  [self.tabBar applyPrimaryThemeWithScheme:self.containerScheme];
+
+  // Then
+  [self verifyTabBarPrimaryTheming];
+}
+
+- (void)verifyTabBarPrimaryTheming {
+  // Color
+  XCTAssertEqualObjects(self.tabBar.barTintColor, self.colorScheme.primaryColor);
+  XCTAssertEqualObjects(self.tabBar.tintColor, self.colorScheme.onPrimaryColor);
+  XCTAssertEqualObjects([self.tabBar titleColorForState:MDCTabBarItemStateSelected],
+                        self.colorScheme.onPrimaryColor);
+  XCTAssertEqualObjects([self.tabBar imageTintColorForState:MDCTabBarItemStateSelected],
+                        self.colorScheme.onPrimaryColor);
+  UIColor *unselectedTitleColor =
+      [self.colorScheme.onPrimaryColor colorWithAlphaComponent:kUnselectedTitleOpacity];
+  UIColor *unselectedImageColor =
+      [self.colorScheme.onPrimaryColor colorWithAlphaComponent:kUnselectedImageOpacity];
```

Comment:
This should be 60%. I believe you were going to fix this in the theming extension and not in the themer?

Reply:
I thought the reason why it calls the themer was something the team decided to do at one time for migration purposes. If that is not the case, a straightforward implementation would actually save more time.

Addressed in 74e741a795f78f626321b5d8bd20850db99dda6c.